### PR TITLE
SEO 2020 meta robots and link additions

### PR DESCRIPTION
**Pull request by Tiggerito**
More changes for #908

### CODE REVIEW

File: sql/2020/07_SEO/pages_wpt_bodies_by_device_and_same_site_link_count.sql [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+  FROM
+    ( 
+      SELECT 
+        _TABLE_SUFFIX AS client,
+        total,
+        get_wpt_bodies_info(JSON_EXTRACT_SCALAR(payload, '$._wpt_bodies')) AS wpt_bodies_info      
+      FROM
+        #`httparchive.pages.2020_08_01_*`
+        `httparchive.sample_data.pages_*`
```
Comment (bazzadp) [inline sql/2020/07_SEO/pages_wpt_bodies_by_device_and_same_site_link_count.sql:61]: Looks like you've left the test tables in here:

```suggestion
        `httparchive.pages.2020_08_01_*`
        #`httparchive.sample_data.pages_*`
```

File: sql/2020/07_SEO/pages_wpt_bodies_by_device_and_same_site_link_count.sql [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        JOIN
+        ( 
+            # to get an accurate total of pages per device. also seems fast
+            SELECT _TABLE_SUFFIX, COUNT(0) AS total 
+            FROM 
+                #`httparchive.pages.2020_08_01_*`
+                `httparchive.sample_data.pages_*`
```
Comment (bazzadp) [inline sql/2020/07_SEO/pages_wpt_bodies_by_device_and_same_site_link_count.sql:68]:
```suggestion
                `httparchive.pages.2020_08_01_*`
                #`httparchive.sample_data.pages_*`
```

**Review by bazzadp — CHANGES_REQUESTED** [review]
These are some big queries you got going on there @Tiggerito !

Spotted a couple of places where you've left the sample tables in.

File: sql/2020/07_SEO/pages_wpt_bodies_by_device_and_same_site_link_count.sql (hunk 1 above)
Comment (Tiggerito) [reply sql/2020/07_SEO/pages_wpt_bodies_by_device_and_same_site_link_count.sql]: I completely removed the sample line. At some point in the past I was told to remove all references to test stuff.

File: sql/2020/07_SEO/pages_wpt_bodies_by_device_and_same_site_link_count.sql (hunk 2 above)
Comment (Tiggerito) [reply sql/2020/07_SEO/pages_wpt_bodies_by_device_and_same_site_link_count.sql]: ditto

File: sql/2020/07_SEO/pages_wpt_bodies_by_device.sql [hunk 3]
```diff
@@ -302,6 +377,9 @@ SELECT
   # Rendering changed Robots
   AS_PERCENT(COUNTIF(wpt_bodies_info.rendering_changed_robots_meta_tag), COUNT(0)) AS pct_rendering_changed_robots_meta_tag,
 
+
+
+
```
Comment (bazzadp) [inline sql/2020/07_SEO/pages_wpt_bodies_by_device.sql:382]: Nit: any reason for all these newlines?

```suggestion
```

File: sql/2020/07_SEO/pages_wpt_bodies_by_device_and_percentile.sql [hunk 4]
```diff
@@ -42,6 +48,15 @@ try {
     
       result.image_links = anchors_rendered.image_links;
       result.text_links = anchors_rendered.text_links;
+
```
Comment (bazzadp) [inline sql/2020/07_SEO/pages_wpt_bodies_by_device_and_percentile.sql:51]: Nit remove at least one of these newlines:

```suggestion
```

File: sql/2020/07_SEO/pages_wpt_bodies_by_device_and_percentile.sql [hunk 5]
```diff
@@ -76,13 +91,22 @@ SELECT
...
   # percent of links are image links
   ROUND(APPROX_QUANTILES(SAFE_DIVIDE(wpt_bodies_info.image_links, wpt_bodies_info.image_links + wpt_bodies_info.text_links), 1000)[OFFSET(percentile * 10)], 4) AS image_links_percent,
 
   # words
   APPROX_QUANTILES(wpt_bodies_info.visible_words_rendered_count, 1000)[OFFSET(percentile * 10)] AS visible_words_rendered,
   APPROX_QUANTILES(wpt_bodies_info.visible_words_raw_count, 1000)[OFFSET(percentile * 10)] AS visible_words_raw
 
+
+
```
Comment (bazzadp) [inline sql/2020/07_SEO/pages_wpt_bodies_by_device_and_percentile.sql:109]: Nit, remove unnecessary newlines:

```suggestion
```

File: sql/2020/07_SEO/pages_wpt_bodies_by_device_and_same_site_link_count.sql [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        ( 
+            # to get an accurate total of pages per device. also seems fast
+            SELECT _TABLE_SUFFIX, COUNT(0) AS total 
+            FROM 
+                `httparchive.pages.2020_08_01_*`
+
+            GROUP BY _TABLE_SUFFIX
+        ) 
+        USING (_TABLE_SUFFIX)
+    )
```
Comment (bazzadp) [inline sql/2020/07_SEO/pages_wpt_bodies_by_device_and_same_site_link_count.sql:71]: Nit. Formatting changes:

```suggestion
  MAX(wpt_bodies_info.links_window_location + wpt_bodies_info.links_window_open + wpt_bodies_info.links_href_javascript) AS max_links_any
FROM ( 
  SELECT 
    _TABLE_SUFFIX AS client,
    total,
    get_wpt_bodies_info(JSON_EXTRACT_SCALAR(payload, '$._wpt_bodies')) AS wpt_bodies_info      
  FROM
    `httparchive.pages.2020_08_01_*`
  JOIN (
    # to get an accurate total of pages per device. also seems fast
    SELECT
      _TABLE_SUFFIX,
      COUNT(0) AS total 
    FROM 
      `httparchive.pages.2020_08_01_*`
    GROUP BY
      _TABLE_SUFFIX
  ) 
  USING
    (_TABLE_SUFFIX)
  )
```

File: sql/2020/07_SEO/pages_wpt_bodies_by_device.sql [hunk 7]
```diff
@@ -368,6 +446,42 @@ SELECT
...
+  AS_PERCENT(COUNTIF(wpt_bodies_info.rendered_googlebot_news_max_snippet), COUNT(0)) AS pct_rendered_googlebot_news_max_snippet,
+  AS_PERCENT(COUNTIF(wpt_bodies_info.rendered_googlebot_news_max_image_preview), COUNT(0)) AS pct_rendered_googlebot_news_max_image_preview,
+  AS_PERCENT(COUNTIF(wpt_bodies_info.rendered_googlebot_news_max_video_preview), COUNT(0)) AS pct_rendered_googlebot_news_max_video_preview,
+  AS_PERCENT(COUNTIF(wpt_bodies_info.rendered_googlebot_news_notranslate), COUNT(0)) AS pct_rendered_googlebot_news_notranslate,
+  AS_PERCENT(COUNTIF(wpt_bodies_info.rendered_googlebot_news_noimageindex), COUNT(0)) AS pct_rendered_googlebot_news_noimageindex,
+  AS_PERCENT(COUNTIF(wpt_bodies_info.rendered_googlebot_news_nocache), COUNT(0)) AS pct_rendered_googlebot_news_nocache,
+
+
   FROM
     (
```
Comment (bazzadp) [inline sql/2020/07_SEO/pages_wpt_bodies_by_device.sql:486]: Nit:

```suggestion
FROM ( 
```

File: sql/2020/07_SEO/pages_wpt_bodies_by_device_and_same_site_link_count.sql [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+GROUP BY
+  client,
+  links_same_site,
+  total
+ORDER BY links_same_site ASC
+LIMIT 100
```
Comment (bazzadp) [inline sql/2020/07_SEO/pages_wpt_bodies_by_device_and_same_site_link_count.sql:77]: To be honest I'm not fully understanding the intent of this query so could be wrong here, but should you really order by number of internal links and also limit by the top 100 with links_same_site?

What if there's some really common internal ink count at 120 links that you're missing out on?

Don't you care more about what the most common number of internal links are?

```suggestion
ORDER BY
  count(0) DESC
LIMIT 100
```

**Review by bazzadp — CHANGES_REQUESTED** [review]
Most of these are formatting nits, but last one is a question as to the logic in the reported data, so interested to hear your opinion on that one @Tiggerito . Let me know if I've picked up the wrong end of stick here and query is fine.

File: sql/2020/07_SEO/pages_wpt_bodies_by_device_and_same_site_link_count.sql (hunk 8 above)
Comment (Tiggerito) [reply sql/2020/07_SEO/pages_wpt_bodies_by_device_and_same_site_link_count.sql:77]: This one has an interesting reason. They want to try and identify SPAs (Single Page Apps) with issues. I came up with the idea of reporting on sites that had few identifiable same site links. So we did want to see just those with very few links.

We're still trying to work this out and learn from the data. I also pulled in the URLs for some zero link pages and there were quite a few different reasons for it.

One was the thing I posted in Slack about. We can't see content including links inside iframes.

I suspect solving the SPA link issue riddle is not going to happen this year. If Google can't work them out, how are we going to!

Comment (bazzadp) [reply sql/2020/07_SEO/pages_wpt_bodies_by_device_and_same_site_link_count.sql:77]: Ah thanks for explaining. Maybe a comment at the top of the query might be useful explaining this?

**Review by bazzadp — APPROVED** [review]

File: sql/2020/07_SEO/pages_wpt_bodies_by_device_and_same_site_link_count.sql (hunk 8 above)
Comment (Tiggerito) [reply sql/2020/07_SEO/pages_wpt_bodies_by_device_and_same_site_link_count.sql:77]: Done

File: sql/2020/07_SEO/pages_wpt_bodies_by_device.sql [hunk 9]
```diff
@@ -168,6 +205,44 @@ try {
...
+      result.rendered_googlebot_news_noarchive = rendered.googlebot_news.noarchive === true;
+      result.rendered_googlebot_news_nosnippet = rendered.googlebot_news.nosnippet === true;
+      result.rendered_googlebot_news_unavailable_after = rendered.googlebot_news.unavailable_after === true;
+      result.rendered_googlebot_news_max_snippet = rendered.googlebot_news.max_snippet === true;
+      result.rendered_googlebot_news_max_image_preview = rendered.googlebot_news.max_image_preview === true;
+      result.rendered_googlebot_news_max_video_preview = rendered.googlebot_news.max_video_preview === true;
+      result.rendered_googlebot_news_notranslate = rendered.googlebot_news.notranslate === true;
+      result.rendered_googlebot_news_noimageindex = rendered.googlebot_news.noimageindex === true;
+      result.rendered_googlebot_news_nocache = rendered.googlebot_news.nocache === true;
+
```
Comment (bazzadp) [inline sql/2020/07_SEO/pages_wpt_bodies_by_device.sql:244]: Nit: unnecessary extra newline:

```suggestion
```

File: sql/2020/07_SEO/pages_wpt_bodies_by_device.sql [hunk 10]
```diff
@@ -368,8 +443,43 @@ SELECT
...
+  AS_PERCENT(COUNTIF(wpt_bodies_info.rendered_googlebot_news_noarchive), COUNT(0)) AS pct_rendered_googlebot_news_noarchive,
+  AS_PERCENT(COUNTIF(wpt_bodies_info.rendered_googlebot_news_nosnippet), COUNT(0)) AS pct_rendered_googlebot_news_nosnippet,
+  AS_PERCENT(COUNTIF(wpt_bodies_info.rendered_googlebot_news_unavailable_after), COUNT(0)) AS pct_rendered_googlebot_news_unavailable_after,
+  AS_PERCENT(COUNTIF(wpt_bodies_info.rendered_googlebot_news_max_snippet), COUNT(0)) AS pct_rendered_googlebot_news_max_snippet,
+  AS_PERCENT(COUNTIF(wpt_bodies_info.rendered_googlebot_news_max_image_preview), COUNT(0)) AS pct_rendered_googlebot_news_max_image_preview,
+  AS_PERCENT(COUNTIF(wpt_bodies_info.rendered_googlebot_news_max_video_preview), COUNT(0)) AS pct_rendered_googlebot_news_max_video_preview,
+  AS_PERCENT(COUNTIF(wpt_bodies_info.rendered_googlebot_news_notranslate), COUNT(0)) AS pct_rendered_googlebot_news_notranslate,
+  AS_PERCENT(COUNTIF(wpt_bodies_info.rendered_googlebot_news_noimageindex), COUNT(0)) AS pct_rendered_googlebot_news_noimageindex,
+  AS_PERCENT(COUNTIF(wpt_bodies_info.rendered_googlebot_news_nocache), COUNT(0)) AS pct_rendered_googlebot_news_nocache,
+
```
Comment (bazzadp) [inline sql/2020/07_SEO/pages_wpt_bodies_by_device.sql:480]:
```suggestion
```

File: sql/2020/07_SEO/pages_wpt_bodies_by_device_and_same_site_link_count.sql [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+links_href_javascript INT64
+ 
+> LANGUAGE js AS '''
+var result = {};
+try {
+  var wpt_bodies = JSON.parse(wpt_bodies_string);
+
+  if (Array.isArray(wpt_bodies) || typeof wpt_bodies != 'object') return result;
+
+ if (wpt_bodies.anchors && wpt_bodies.anchors.rendered) {
```
Comment (bazzadp) [inline sql/2020/07_SEO/pages_wpt_bodies_by_device_and_same_site_link_count.sql:25]:
```suggestion
  if (wpt_bodies.anchors && wpt_bodies.anchors.rendered) {
```

File: sql/2020/07_SEO/pages_wpt_bodies_by_device_and_same_site_link_count.sql [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+      _TABLE_SUFFIX
+  ) 
+  USING
+    (_TABLE_SUFFIX)
+  )
+GROUP BY
+  client,
+  links_same_site,
+  total
+ORDER BY links_same_site ASC
```
Comment (bazzadp) [inline sql/2020/07_SEO/pages_wpt_bodies_by_device_and_same_site_link_count.sql:77]:
```suggestion
ORDER BY
  links_same_site ASC
```

**Review by bazzadp — APPROVED** [review]
Some more spacing nits that I'm just gonna go ahead and commit and then merge this.

**Pull request merged**